### PR TITLE
Change remote methods to resource methods

### PR DESCRIPTION
**Pull request by ThishaniLucas**
| Existing           | Changed to            | Sample usage                                                                |
|--------------------|-----------------------|-----------------------------------------------------------------------------|
|                    |                       |                                                                             |
| getById            | get /read/id          | fhirConnectorObj->/read/id(resType, id);                                    |
| getByVersion       | get /read/version     | fhirConnectorObj->/read/'version(resType, id = "123", 'version = 'version); |
| update             | put /fhirResource     | fhirConnectorObj->/fhirResource.put(resPayload);                            |
| patch              | patch /fhirResource   | fhirConnectorObj->/fhirResource.patch(resType, id, resPayload)              |
| delete             | delete /fhirResource  | fhirConnectorObj->/fhirResource.delete(resType, id);                        |
| getInstanceHistory | get /history/instance | fhirConnectorObj->/history/instance("Patient", "1234", {"_count": "10"});   |
| create             | post /fhirResource    | fhirConnectorObj->/fhirResource.post(resPayload);                           |
| search             | get /search/type      | fhirConnectorObj->/search/'type(resType, request.getQueryParams());         |
| getHistory         | get /history/type     | fhirConnectorObj->/history/'type("Patient", {_count: "10"});                |
| getConformance     | get /metadata         | fhirConnectorObj->/metadata();                                              |
| getAllHistory      | get /history          |                                                                             |
| searchAll          | get /search           |                                                                             |
| batchRequest       | post /batch           |                                                                             |
| transaction        | post /transaction     |                                                                             |
| nextBundle         | get /bundle/next      |                                                                             |

### CODE REVIEW

File: fhir/fhir_connector.bal [hunk 1]
```diff
@@ -71,7 +71,7 @@ public isolated client class FHIRConnector {
     # + summary - The subset of the resource content to be returned
     # + return - If successful, FhirResponse record else FhirError record
     @display {label: "Get resource by the logical ID"}
-    remote isolated function getById(@display {label: "Resource Type"} ResourceType|string 'type,
+    isolated resource function get read/id(@display {label: "Resource Type"} ResourceType|string 'type,
```
Comment (rdulmina) [inline fhir/fhir_connector.bal:74]: If we don't need annotation to be attached to param `id` we can rewrite this method using path params as `isolated resource function get read/[string id](@display {label: ...` See https://ballerina.io/learn/by-example/http-path-param/

We can call this method like `check fhirConnector->/read/pat1(PATIENT);`

If `pat1` is a variable we can also do
```bal
string id = "pat1";
check fhirConnector->/read/[id](PATIENT);
```

Comment (rdulmina) [reply fhir/fhir_connector.bal:74]: Lets update similar places as well

**Review by nirmal070125 — CHANGES_REQUESTED** [review]
The changes to the operation names are still in discussion. Hold merging till we hear from the ballerina team.

**Comment by sameerajayasoma** [comment]
I would use `fhir` or `fhir-resource` instead of `fhirResource` in the resource path.

**Pull request closed without merging**